loan-events/src/agent.spec.js

```diff
@@ -0,0 +1,255 @@
+  });
+
+  it('protocolAbbreviation key required', () => {
+    const { protocolAbbreviation } = config;
+    expect(typeof (protocolAbbreviation)).toBe('string');
+    expect(protocolAbbreviation).not.toBe('');
+  });
+
+  it('developerAbbreviation key required', () => {
+    const { developerAbbreviation } = config;
+    expect(typeof (developerAbbreviation)).toBe('string');
+    expect(developerAbbreviation).not.toBe('');
+  });
+
+  it('contracts key required', () => {
+    const { contracts } = config;
+    expect(typeof (contracts)).toBe('object');
+    expect(contracts).not.toBe({});
+  });
+
+  it('contracts key values must be valid', () => {
+    const { contracts } = config;
+    Object.keys(contracts).forEach((key) => {
+      const { address, abiFile, events } = contracts[key];
+
+      // only check that an address exists for the LendingPoolAddressesProvider entry
+      if (key === 'LendingPoolAddressesProvider') {
+        expect(address).not.toBe(undefined);
+        
+        // check that the address is a valid address
+        expect(ethers.utils.isHexString(address, 20)).toBe(true);
+      }
+
+      // load the ABI from the specified file
+      // the call to getAbi will fail if the file does not exist
+      const abi = utils.getAbi(abiFile);
+
+      if (events !== undefined) {
+        const eventObjects = getObjectsFromAbi(abi, 'event');
+        // for all of the events specified, verify that they exist in the ABI
+        Object.keys(events).forEach((eventName) => {
+          expect(Object.keys(eventObjects).indexOf(eventName)).not.toBe(-1);
+
+          const entry = events[eventName];
+          const { type, severity } = entry;
+
+          // check type, this will fail if 'type' is not valid
+          expect(Object.prototype.hasOwnProperty.call(FindingType, type)).toBe(true);
+
+          // check severity, this will fail if 'severity' is not valid
+          expect(Object.prototype.hasOwnProperty.call(FindingSeverity, severity)).toBe(true);
+        });
+      }
+    });
+  });
+});
+
+// tests
+describe('monitor emitted events', () => {
+  describe('handleTransaction', () => {
+    let initializeData;
+    let developerAbbreviation;
+    let protocolAbbreviation;
+    let protocolName;
+    let handleTransaction;
+    let mockTxEvent;
+    let iface;
+    let abi;
+    let eventInConfig;
+    let eventNotInConfig;
+    let findingType;
+    let findingSeverity;
+    const mockContractAddress = ethers.utils.getAddress('0x230E76a625927709618AB27761d574A5CFf75d61');
+    const contractName = 'LendingPool';
+
+    beforeEach(async () => {
+      initializeData = {};
+
+      mockContract.getLendingPool = jest.fn().mockResolvedValue(mockContractAddress);
+
+      // initialize the handler
+      await (provideInitialize(initializeData))();
+      handleTransaction = provideHandleTransaction(initializeData);
+
+      ({ protocolName, protocolAbbreviation, developerAbbreviation } = config);
```

Comment:
nit, but these parens are superfluous

```suggestion
      { protocolName, protocolAbbreviation, developerAbbreviation } = config;
```

Reply:
These parens are required when destructuring using variables that have already been declared.